cloudinit/url_helper.py

```diff
@@ -639,16 +639,22 @@ def oauth_headers(
     return signed_headers
 
 
-def retry_on_url_exc(msg, exc):
-    """readurl exception_cb that will retry on NOT_FOUND and Timeout.
+def retry_on_url_exc(
+    msg, exc, *, retry_codes=(NOT_FOUND,), retry_instances=(requests.Timeout,)
```

Comment:
Azure is the only user of this callback I think, but it seems reasonable that ConnectionError should also be included for retry.